### PR TITLE
SOLR-16281: review boxing unboxing

### PR DESCRIPTION
**Pull request by epugh**
https://issues.apache.org/jira/browse/SOLR-16281

# Description

boxing/unboxing has been improved since Java 5!

# Solution

Apply suggestiosn!

# Tests

Please describe the tests you've developed or run to confirm this patch implements the feature or solves the problem.

# Checklist

Please review the following and check all that apply:

- [ ] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [ ] I have created a Jira issue and added the issue ID to my pull request title.
- [ ] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [ ] I have developed this patch against the `main` branch.
- [ ] I have run `./gradlew check`.
- [ ] I have added tests for my changes.
- [ ] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

### CODE REVIEW

File: solr/core/src/test/org/apache/solr/handler/component/TestPivotHelperCode.java [hunk 1]
```diff
@@ -89,8 +89,7 @@ public void testCompareWithNullLast() {
     assertEquals(a.compareTo(b), PivotFacetFieldValueCollection.compareWithNullLast(a, b));
     assertEquals(b.compareTo(a), PivotFacetFieldValueCollection.compareWithNullLast(b, a));
 
-    Long bb = new Long(b.longValue());
-    assertEquals(0, PivotFacetFieldValueCollection.compareWithNullLast(b, bb));
+    assertEquals(0, PivotFacetFieldValueCollection.compareWithNullLast(b, b));
```
Comment (cpoerschke) [inline solr/core/src/test/org/apache/solr/handler/component/TestPivotHelperCode.java:92]: hmm, not sure about this one, looks like it's part of the test?

**Review by cpoerschke — COMMENTED** [review]
29 / 29 files viewed

File: solr/core/src/test/org/apache/solr/handler/component/TestPivotHelperCode.java (hunk 1 above)
Comment (epugh) [reply solr/core/src/test/org/apache/solr/handler/component/TestPivotHelperCode.java:92]: As I edited it, I wondered if you were going to call this one out!  Eagle Eyes!    I could totally be swayed to undo it.     The line `Long bb = new Long(b.longValue()); ` means we are basically testing Java functionality???

Comment (epugh) [reply solr/core/src/test/org/apache/solr/handler/component/TestPivotHelperCode.java:92]: you know what, I'll just roll this cahnge back ;-)

**Comment by epugh** [comment]
Do you think this is ready now for merging @cpoerschke ??

**Review by cpoerschke — APPROVED** [review]

**Pull request merged**